funfact/backend/_torch.py

```diff
@@ -14,16 +14,20 @@ class PyTorchBackend(metaclass=BackendMeta):
     tensor_t = (torch.Tensor, np.ndarray)
 
     @classmethod
-    def tensor(cls, array, **kwargs):
-        return torch.tensor(array, **kwargs)
+    def tensor(cls, array, optimizable, **kwargs):
+        return torch.tensor(array, requires_grad=optimizable, **kwargs)
 
     @classmethod
     def seed(cls, key):
         cls._gen.manual_seed(key)
 
     @classmethod
-    def normal(cls, mean, std, *shape, dtype=torch.float32):
-        return torch.normal(mean, std, shape, dtype=dtype, generator=cls._gen)
+    def normal(cls, mean, std, *shape, optimizable, dtype=torch.float32):
+        data = torch.normal(mean, std, shape, dtype=dtype, generator=cls._gen)
+        if optimizable:
+            return data.clone().detach().requires_grad_(True)
```

Comment:
This could cause some unnecessary overhead due to `clone`. Can we directly override the `requires_grad` property?

Reply:
This is a solution I found somewhere on a forum. We don't want to set the `requires_grad` property before the random data is generated as then the tracing could include the rng method.

This is a relevant thread: https://stackoverflow.com/questions/51751231/pytorch-effect-of-normal-initialization-on-gradients

Maybe a solution could be to create an `as_tensor` method that can create torch `tensor`s from numpy array and use the numpy rng in the `normal` function for the torch backend.

Something like:
```
def normal(cls, mean, std, *shape, optimizable, dtype=torch.float32):
    data = np_backend.normal(cls, mean, std, *shape)
    return as_tensor(data, optimizable=optimizable, dtype=dtype)
```

What do you think?

Reply:
Yeah, honestly I also recall reading this quirk with PyTorch a long while ago. The numpy solution does not seem very attractive either :rofl: I guess we can just settle with what it is for now.